### PR TITLE
fix: IconCardGroup generic typing

### PR DESCRIPTION
**Pull request by 0ces**
This is a fix to be able to use the IconCardGroup with enum values

### CODE REVIEW

File: src/inputs/IconCardGroup.tsx [hunk 1]
```diff
@@ -41,14 +44,35 @@ export function IconCardGroup(props: IconCardGroupProps) {
...
+  const [selected, setSelected] = useState<V[]>(values);
+
+  const toggleValue = useCallback(
+    (value: V) => {
+      if (selected.includes(value)) {
+        setSelected(selected.filter((v) => v !== value));
+      } else {
+        setSelected([...selected, value]);
+      }
+      onChange([...selected, value]);
```
[the remark this answers is not in the document]
Comment (0ces) [reply src/inputs/IconCardGroup.tsx:59]: You're right, I'll fix it

File: src/inputs/IconCardGroup.tsx [hunk 2]
```diff
@@ -9,29 +7,34 @@ import { useTestIds } from "src/utils";
...
-export interface IconCardGroupItemOption {
+export interface IconCardGroupItemOption<V extends Value> {
   icon: IconProps["icon"];
   label: string;
   disabled?: boolean;
   /** The value of the IconCardGroup item, stored in value array in state. */
-  value: string;
+  value: V;
+  /** Exclusive: if true, this option will override all other options when selected. */
+  exclusive?: boolean;
```
Comment (JonnCharpentier) [inline src/inputs/IconCardGroup.tsx:21]: question: Do we need this 'exclusive' param?

Comment (0ces) [reply src/inputs/IconCardGroup.tsx:20]: This is meant for options that need to be exclusive like "N/A"

**Review by JonnCharpentier — APPROVED** [review]
LGTM!

**Pull request merged**